src/compiler/checker.ts

```diff
@@ -32436,7 +32440,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
         for (let i = 0; i < argCount; i++) {
             const arg = args[i];
-            if (arg.kind !== SyntaxKind.OmittedExpression && !(checkMode & CheckMode.IsForStringLiteralArgumentCompletions && hasSkipDirectInferenceFlag(arg))) {

```

Comment:
This change is quite nice (IMHO) because it eliminates the need for this special `CheckMode` almost entirely (in fact, the first commit in this PR does exactly that - it just doesn't fix the issue).

With this we can fully rely *just* on "node blocking" to alter the default inference behavior for the completion purposes.